# Ground State Zinc Oxide

In [1]:
# Give some estimate for MT radii, maintaining a consistent ratio between T and O
from src.materials import ZnOWurzite
from src.optimal_muffin_tins import main

# ZnO in angstrom, implying these bond lengths are too
main(ZnOWurzite, scaling_factor=0.9)

Minimum bond length between atomic numbers (8, 8): 3.2077121269530924
Minimum bond length between atomic numbers (8, 30): 1.9738161289053884
For each bond, compute the MT radius associated with the element with the smallest MT radius. 
Do this for each bond, and choose the minimum.
[1.6039, 0.8868]
Smallest muffin tin radius for element 8 is 0.79812, scaled by 0.9
Use this to determine the MT radii for all other elements in the system, with optimal ratio 
determined by tabulated rkgmax ratios
Atomic Number 8, MT radius 0.79812, sum of MT radii 1.59624, min_bond_length 3.2077121269530924
Atomic Number 30, MT radius 0.9783467781935792, sum of MT radii 1.7764667781935792, min_bond_length 1.9738161289053884


#### MT Selection Summary

Smallest muffin tin radius in the system is for element 8: 0.79812 (angstrom, scaling factor = 0.9)
Atomic Number 30, MT radius 0.9783467781935792 (angstrom scaling factor = 0.9)

So in Bohr, try MT_O = 1.50 and MT_Zn = 1.85

Dima's paper. DOI: 10.1103/PhysRevB.94.035118
His MTs are: 1.6 angstrom for O and Zn.
In the case of binary materials, we use equal RMT for both species.

In [11]:
# Lattice vectors from lattice constants
import ase
import numpy as np

# In angstrom
# This looks right, but one should multiply through by -1
lattice_returned_by_pymatgen = np.array([[-1.624700, -2.814063, 0.000000], [-1.624700, 2.814063, 0.000000], [0.000000, 0.000000, -5.203800]])

# Straight from the cif - angstrom and degrees
cell = ase.geometry.cell.Cell.fromcellpar([3.249400, 3.249400, 5.203800, 90, 90, 120])

# http://aflowlib.org/prototype-encyclopedia/trigonal_lattice.html#lattice10
print(cell.get_bravais_lattice())
print(vars(cell))

# In angstrom ... this really doesn't look right, despite ASE saying it's a hexagonal lattice
lattice_returned_by_ase = np.array([[3.2494,  0.,  0.], [-1.6247,  2.81406295,  0. ], [0.,  0.,  5.2038]])

# zno
#  pymatgen lattice vectors and positions - I assume they are consistent, at least.
# zno_alt
#  Positions from the cif, but unsure what to set the lattice vectors as.


HEX(a=3.2494, c=5.2038)
{'array': array([[ 3.2494    ,  0.        ,  0.        ],
       [-1.6247    ,  2.81406295,  0.        ],
       [ 0.        ,  0.        ,  5.2038    ]])}
